inst/stan/epinowcast.stan

```diff
@@ -163,48 +175,92 @@ model {
       rd_eff_sd ~ zero_truncated_normal(rd_eff_sd_p[1], rd_eff_sd_p[2]);
     } 
   }
+  // share of cases with eventually known reference date
+  alpha_sd ~ normal(alpha_sd_p[1], alpha_sd_p[2]) T[0, ];
+  for (k in 1:g){
+    alpha_start[k] ~ normal(alpha_start_p[1], alpha_start_p[2]);
+    alpha_epsilon[k] ~ normal(0, alpha_sd);
+  }
   // reporting overdispersion (1/sqrt)
   sqrt_phi ~ normal(sqrt_phi_p[1], sqrt_phi_p[2]) T[0,];
   }
   // log density: observed vs model
   if (likelihood) {
     profile("model_likelihood") {
-    target += reduce_sum(obs_lupmf, st, 1, flat_obs, sl, csl, imp_obs, sg, st,
-                         rdlurd, srdlh, ref_lh, dpmfs, ref_p, phi);
+    target += obs_lupmf(flat_obs | obs_miss, dmax, sl, csl, g, imp_obs, sg, st,
+                         rdlurd, srdlh, ref_lh, dpmfs, ref_p, alpha, phi);
     }
   }
 }
 
 generated quantities {
   int pp_obs[pp ? sum(sl) : 0];
+  int pp_obs_miss[pp ? sum(sl) : 0];
   vector[ologlik ? s : 0] log_lik;
+  vector[ologlik ? (t-dmax) : 0] log_lik_miss;
   int pp_inf_obs[cast ? dmax : 0, cast ? g : 0];
+  int pp_inf_obs_miss[cast ? t : 0, cast ? g : 0];
+  int pp_inf_obs_miss_rep[cast ? (t-dmax) : 0, cast ? g : 0];
   profile("generated_total") {
   if (cast) {
-    real tar_obs;
-    vector[dmax] exp_obs;
+    int i_group, i_time;
+    real tar_obs, tar_alpha;
     vector[dmax] rdlh;
+    vector[dmax] exp_obs;
+    vector[ologlik ? t : 0] exp_obs_miss_rep[ologlik ? g : 0];
     int pp_obs_tmp[s, dmax];
+    int pp_obs_tmp_miss[s, dmax];
+    exp_obs_miss_rep = rep_array(rep_vector(0, t), g);
+    pp_inf_obs_miss_rep = rep_array(0, (t-dmax), g);
     // Posterior predictions for observations
     for (i in 1:s) {
       profile("generated_obs") {
-      tar_obs = imp_obs[sg[i]][st[i]];
-      rdlh = srdlh[rdlurd[st[i]:(st[i] + dmax - 1), sg[i]]];
+      i_group = sg[i];
+      i_time = st[i];
+      // estimated expected cases
+      tar_obs = imp_obs[i_group][i_time];
+      // estimated share of known reference dates
+      tar_alpha = alpha[i_group][i_time];
+      // reference date delay hazard
+      rdlh = srdlh[rdlurd[i_time:(i_time + dmax - 1), i_group]];
+      // expected observations with delay 0:(dmax-1)
       exp_obs = expected_obs(tar_obs, ref_lh[1:dmax, dpmfs[i]], rdlh, ref_p);
-      pp_obs_tmp[i, 1:dmax] = neg_binomial_2_rng(exp_obs, phi);
+      // realized observations with known reference date
+      pp_obs_tmp[i, 1:dmax] = neg_binomial_2_rng(exp_obs * tar_alpha, phi);
+      // realized observations with unknown reference date
+      pp_obs_tmp_miss[i, 1:dmax] = neg_binomial_2_rng(exp_obs * (1 - tar_alpha), phi);
       }
       profile("generated_loglik") {
       if (ologlik) {
+        // predicted observations with missing reference date by reporting date
+        if (i_time > 1) {
+          exp_obs_miss_rep[i_group][i_time:min(i_time + dmax - 1, t)]
+            += (exp_obs * (1 - tar_alpha))[1:min(dmax, t - i_time + 1)];
```

Comment:
In many places (e.g. here) I am assuming that `dmax<=t`. Otherwise, the model would break. Is this a reasonable assumption/requirement? And if yes, should we add a check for this somewhere?

Reply:
Hmm so I think that we don't want to assume this for the general model but it seems like we need to for the missingness model? Given that I would guess we need to specify a check when a missingness model is asked for?